resdk/resources/collection.py

```diff
@@ -264,3 +264,11 @@ def remove_samples(self, *samples):
             self.resolwe.api.sample(sample).remove_from_collection.post({'ids': [self.id]})
 
         self.samples.clear_cache()
+
+    def duplicate(self):
+        """Duplicate (make copy of) ``collection`` object."""
+        if self.id is None:
+            raise ValueError('Instance must be saved before making a duplicate.')
+
+        duplicated = self.resolwe.api.collection().duplicate.post({'ids': [self.id]})
+        return duplicated[0]
```

Comment:
You need to return a `Collection` object, something like `self.__class__(resolwe=self.resolwe, **duplicated[0])`